presto-iceberg/src/main/java/com/facebook/presto/iceberg/FilesTable.java

```diff
@@ -121,53 +126,58 @@ private static List<Page> buildPages(ConnectorTableMetadata tableMetadata, Table
         TableScan tableScan = getTableScan(TupleDomain.all(), snapshotId, icebergTable).includeColumnStats();
         Map<Integer, Type> idToTypeMap = getIdToTypeMap(icebergTable.schema());
 
-        tableScan.planFiles().forEach(fileScanTask -> {
-            DataFile dataFile = fileScanTask.file();
-            pagesBuilder.beginRow();
-            pagesBuilder.appendInteger(dataFile.content().id());
-            pagesBuilder.appendVarchar(dataFile.path().toString());
-            pagesBuilder.appendVarchar(dataFile.format().name());
-            pagesBuilder.appendBigint(dataFile.recordCount());
-            pagesBuilder.appendBigint(dataFile.fileSizeInBytes());
-            if (checkNonNull(dataFile.columnSizes(), pagesBuilder)) {
-                pagesBuilder.appendIntegerBigintMap(dataFile.columnSizes());
+        try (CloseableIterable<FileScanTask> fileScanTasks = tableScan.planFiles()) {
```

Comment:
Maybe we should also set `fileScanTasks` to `CloseableIterable.empty()` considering the memory leak problem described in #21612. Also refer to trino: https://github.com/trinodb/trino/blob/master/plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergSplitSource.java#L362

Reply:
I don't think we need to do this here because `fileScanTasks` is a temporary variable and will be dropped and cleaned up by the GC soon after this method ends. In the linked issue the reference to the split source is long lived and so without dropping the reference to the original iterable the GC won't properly reap the references